datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAttachmentPickerViewHolderFactoryEspressoTest.kt

```diff
@@ -376,6 +376,105 @@ class QuestionnaireItemAttachmentPickerViewHolderFactoryEspressoTest {
       .isEqualTo("Document File")
   }
 
+  @Test
+  fun shouldNotDisplayPreviewFromNullAnswer() {
+    val questionnaireItemView =
+      QuestionnaireItemViewItem(
+        Questionnaire.QuestionnaireItemComponent().apply {
+          addExtension().apply {
+            url = "http://hl7.org/fhir/StructureDefinition/mimeType"
+            setValue(CodeType("image/*"))
+          }
+          addExtension().apply {
+            url = "http://hl7.org/fhir/StructureDefinition/mimeType"
+            setValue(CodeType("application/pdf"))
+          }
+        },
+        QuestionnaireResponse.QuestionnaireResponseItemComponent().apply { addAnswer(null) },
+        validationResult = NotValidated,
+        answersChangedCallback = { _, _, _ -> },
+      )
+
+    runOnUI { viewHolder.bind(questionnaireItemView) }
+
+    assertThat(viewHolder.itemView.findViewById<ImageView>(R.id.photo_preview).visibility)
+      .isEqualTo(View.GONE)
+
+    assertThat(viewHolder.itemView.findViewById<LinearLayout>(R.id.file_preview).visibility)
+      .isEqualTo(View.GONE)
+
+    assertThat(viewHolder.itemView.findViewById<Button>(R.id.delete).visibility)
+      .isEqualTo(View.GONE)
+  }
+
+  @Test
+  fun shouldNotDisplayPreviewFromNullAnswerAfterPreviouslyBindingNonNullAnswer() {
```

Comment:
optional : may be we can name something like
```suggestion
  fun doNotShowPreviewOfPreviousAnswerAttachmentForCurrentAnswerItem() {
```

Reply:
sure